src/profilePage/ProfilePage.tsx

```diff
@@ -1,22 +1,200 @@
+import { useQuery } from "@tanstack/react-query";
+import { useEffect, useState } from "react";
 import Calendar from "react-github-contribution-calendar";
+import { octokit } from "../../environment/apiKey";
 import {
   monthLabelAttributes,
   panelAttributes,
   panelColors,
   until,
-  values,
   weekLabelAttributes,
 } from "../Models/calendarHeatmap";
+import {
+  CommitData,
+  CommitItem,
+  LanguageData,
+  ProfileItem,
+  RepoItem,
+} from "../Models/interfaces";
 import Drawer from "../Navigation/Drawer";
+import { Route } from "../routes";
 import QuickStats from "./QuickStats";
 import RepoCard from "./RepoCard";
 import StatsCarousel from "./StatsCarousel";
 import StatsRadial from "./StatsRadial";
 
+interface RouteParams {
+  profileId: string;
+}
+
 function ProfilePage() {
+  const [profile, setProfile] = useState<ProfileItem | null>(null);
+  const [repos, setRepos] = useState<RepoItem[]>([]);
+  const [repoNumber, setRepoNumber] = useState<number>(0);
+  const [commitsNumber, setCommitsNumber] = useState<number>(0);
+  const [dataReady, setDataReady] = useState(false);
+  const [result, setResult] = useState<LanguageData>({});
+  const [commitsReady, setCommitsReady] = useState(false);
+  const [commits, setCommits] = useState<CommitData>({});
+
+  const { profileId } = Route.useParams<RouteParams>();
+  const profileName = profileId;
+
+  useEffect(() => {
+    getProfile();
+    getRepos();
+  }, []);
+
+  const getProfile = async () => {
+    if (profileName) {
+      const res = await octokit.request(
+        `GET https://api.github.com/users/${profileName}`
+      );
+
+      if (res.status === 200) {
+        const profile = {
+          name: res.data.name,
+          bio: res.data.bio,
+          login: res.data.login,
+          avatar_url: res.data.avatar_url,
+          followers: res.data.followers,
+          following: res.data.following,
+          organizations_url: res.data.organizations_url,
+        };
+        setProfile(profile);
+        return profile;
+      } else {
+        console.error("Request failed with status:", res.status);
+        return;
+      }
+    }
+  };
+
+  const getRepos = async () => {
+    if (profileName) {
+      try {
+        const res = await octokit.request(
+          `GET https://api.github.com/users/${profileName}/repos?sort=updated`
+        );
+
+        if (res.status === 200) {
+          let repos = res.data.length;
+          const data = res.data;
+          const newRepos = data.map((repo: RepoItem) => ({
```

Comment:
Please move `getProfile` and `getRepos` to a separate file, `api.ts` which you can store in, for example, a `services` directory inside `src`.